src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -319,34 +318,57 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // For Control frames Data Size value shall be less than MTU header
   // For Single and Consecutive Data Size value shall be greater than 0x00
   // and shall be less than payload size
-  if (header.dataSize > payload_size) {
-    LOG4CXX_WARN(logger_,
-                 "Packet data size is "
-                     << header.dataSize
-                     << " and bigger than allowed payload size " << payload_size
-                     << " bytes");
-    return RESULT_FAIL;
-  }
+  // For First Frame Data Size shall be equal 0x08 (payload of this packet will
+  // be 8 bytes).
+
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
-      if (header.dataSize <= 0u) {
+      if (header.dataSize <= 0u || header.dataSize > payload_size) {
```

Comment:
@KVGrygoriev as I could see `dataSize` is `uint32` so you can remove this check "less than zero"

Reply:
@AKalinich-Luxoft what it changes?
I agree, but dataSize for example can be changed to int32_t for some reasons.

Reply:
@EKuliiev it removes redundant check that uint32 less than zero.
Even if there will be some reasons to change type to `int32_t`, you must analyze whole affected area where updated variable is used. Including these range checks. Isn't it?